packages/vite/src/shared/moduleRunnerTransport.ts

```diff
@@ -232,17 +224,18 @@ export const normalizeModuleRunnerTransport = (
           },
         }
       : {}),
-    async send(data) {
+    send(data) {
       if (!invokeableTransport.send) return
 
       if (!isConnected) {
-        if (connectingPromise) {
-          await connectingPromise
-        } else {
+        if (!connectingPromise) {
           throw new Error('send was called before connect')
         }
+        connectingPromise.then(() => {
+          invokeableTransport.send!(data)
+        })
```

Comment:
For `send`s that were called while connecting the error would be voided and won't be able to catch by the user. In Vite 5, the error happened for the `send` call later that was called (which is confusing I guess).
https://github.com/vitejs/vite/blob/c54c860f9d90e4074e5321648f9c5ee9fbda7038/packages/vite/src/shared/hmr.ts#L180-L190